llmfoundry/tokenizers/tiktoken.py

```diff
@@ -101,6 +112,24 @@ def get_vocab(self) -> Dict[str, int]:
             except KeyError:
                 pass
 
+        # As far as I can tell, we don't require get_vocab to completely work,
+        # but when using additional_special_tokens, Hugging Face determines the next
+        # token index to add with len(self.get_vocab()) so we need the _size_ of this dictionary to be correct.
+        extra_id_index = 0
+        candidate_extra_id = f'<extra_id_{extra_id_index}>'
+        indices_to_fill_in = {i for i in range(self.vocab_size)} - set(
```

Comment:
for my understanding, self.vocab_size > vocab.values() ?

Reply:
tbh i don't know exactly what each is supposed to correspond to (and may be different per tokenizer), but in this case, vocab_size is what tiktoken reports the index of its largest index token to be, and vocab.values() just contains all the tokens we could decode on their own. so it excludes the gap in token indices with tiktoken, and any token indices that cannot be converted to a string on their own.